File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.go
func maximumDifference(nums []int) int {
    n := len(nums)

    maxDiff := -1
    minNum := nums[0]

    for j := 1; j < n; j++ {
        if minNum < nums[j] {
            diff := nums[j] - minNum
            if diff > maxDiff {
                maxDiff = diff
            }
        }
        if nums[j] < minNum {
            minNum = nums[j]
        }
    }

    return maxDiff
}

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int n = nums.length;

        int maxDiff = -1;
        int minNum = nums[0];

        for(int j = 1; j < n; j++) {
            if(minNum < nums[j]) {
                maxDiff = Math.max(maxDiff, nums[j] - minNum);
            }
            minNum = Math.min(minNum, nums[j]);
        }

        return maxDiff;
    }
}